Model files

DanielBachler opened the pull request:
The model files are ready for review, then merging into development.  This should now allow for easier sending and receiving of information from the database.

Commit:
Add new files for model files

Commit:
Model files for all existing "documents" made

The model files and generated model files are completed for the current iteration of the database.

Commit:
Add function to send back to firestore.

Add the toFirestore() method that sends data back to firestore from the app.

Commit:
Fix build error

The build would fail due to calling new User.fromFirebase with only 2 args, need 3 args.  Added Temporary blank map to allow for build, will need replacing later.

SpencerCornish commented on lib/src/model/meal.dart:
```diff
@@ -0,0 +1,43 @@
+  Meal._();
+  factory Meal([updates(MealBuilder b)]) = _$Meal;
+
+  factory Meal.frmFirebase(Map<String, dynamic> firestoreData) =>
```
from is spelled wrong

SpencerCornish commented on lib/src/model/meal.dart:
```diff
@@ -0,0 +1,43 @@
+      ..uid = firestoreData['uid']
+      ..startTime = DateTime.parse(firestoreData['start_time'])
+      ..endTime = DateTime.parse(firestoreData['end_time'])
+      ..menu =firestoreData['menu']
```
This is going to come in as an array, so you're going to have to do a `new BuiltList.from(the array)`, since a BuiltList isn't the same as an array

SpencerCornish commented on lib/src/model/meal.dart:
```diff
@@ -0,0 +1,43 @@
+    'uid':uid,
+    'start_time':startTime.toIso8601String(),
+    'end_time':endTime.toIso8601String(),
+    'menu':menu,
```
Same thing here, you'll need to turn this back into an array

SpencerCornish commented on lib/src/model/meal.dart:
```diff
@@ -0,0 +1,43 @@
+
+import 'package:built_value/built_value.dart';
+import 'package:built_collection/built_collection.dart';
+import 'package:firebase/firebase.dart' as fb;
```
Is this used?

SpencerCornish commented on lib/src/model/activity.dart:
```diff
@@ -0,0 +1,56 @@
+
+import 'package:built_value/built_value.dart';
+import 'package:built_collection/built_collection.dart';
+import 'package:firebase/firebase.dart' as fb;
```
Is this still used?

SpencerCornish commented on lib/src/model/user.dart:
```diff
@@ -13,45 +13,95 @@ abstract class User implements Built<User, UserBuilder> {
+  String get phoneNumber;
 
-  // /// [cellNumber] may be an empty string
-  // String get cellNumber;
```
They also have a cell number listeed on the CSV of the userdata they sent.

SpencerCornish commented on lib/src/model/user.dart:
```diff
@@ -13,45 +13,95 @@ abstract class User implements Built<User, UserBuilder> {
-  // /// [dietaryRestrictions] is a comma separated list of restricted dietary items for the user
-  // String get dietaryRestrictions;
+  /// [emergencyContacts] is a built list of [EmergencyContact] objects
+  BuiltList<String> get emergencyContacts;
```
`EmergencyContact` Objects should be another model most likely

Commit:
Merge branch 'development' of https://github.com/SpencerCornish/belgrade-senior-center into model-files

Commit:
Changes from compiling

Commit:
Resolve issues from PR

Commit:
Add EmergencyContact model file

Add a new emergencyContact class that will be used to hold the emergency contacts of the clients.

SpencerCornish committed:
fix spelling, format
